packages/editor/src/components/post-trash/index.js

```diff
@@ -18,18 +18,23 @@ import PostTrashCheck from './check';
 /**
  * Displays the Post Trash Button and Confirm Dialog in the Editor.
  *
+ * @param {?{onActionPerformed: Object}} An object containing the onActionPerformed function.
  * @return {JSX.Element|null} The rendered PostTrash component.
  */
-export default function PostTrash() {
-	const { isNew, isDeleting, postId, title } = useSelect( ( select ) => {
-		const store = select( editorStore );
-		return {
-			isNew: store.isEditedPostNew(),
-			isDeleting: store.isDeletingPost(),
-			postId: store.getCurrentPostId(),
-			title: store.getCurrentPostAttribute( 'title' ),
-		};
-	}, [] );
+export default function PostTrash( { onActionPerformed } ) {
+	const { isNew, isDeleting, postId, title, item } = useSelect(
+		( select ) => {
+			const store = select( editorStore );
+			return {
+				isNew: store.isEditedPostNew(),
+				isDeleting: store.isDeletingPost(),
+				postId: store.getCurrentPostId(),
+				title: store.getCurrentPostAttribute( 'title' ),
+				item: store.getCurrentPost(),
```

Comment:
Can we avoid calling the selector here which triggers a pre-rendering on each range and only call `getCurrentPost` within the callback.

Reply:
Hi @youknowriad good catch, the suggestion was applied.